### DISCUSSION TITLE
Firestore Geofirestore

### DISCUSSION THREAD

**Discussion by mnurulhuda** [invertase/react-native-firebase#4382] [Q&A] [answered]
Please help me.
I need to get one user with the closest location from Cloud Firestore.
Can I use the geofirestore with this react-native-firebase ?

**Comment by mikehardy** [comment, marked as answer]
react-native-firebase is based on firebase-android-sdk and firebase-ios-sdk for the two platforms respectively

geofirestore is based on firebase-js-sdk, so will use a completely different mechanism to communicate with the firestore database in the cloud

It is possible to use both at the same time, so it is possible to for it to work, but you may have data coherency issues if both are attempting to cache things and have slightly different data. Or it might work great.

react-native-firebase attempts to mimic the firebase-js-sdk API so it looks familiar, even though it is using the native SDKs underneath things, so they will look almost completely compatible but you will need to treat each library as if they are completely separate in your app and *not* mix API calls between them I think, for things to work well.

**Comment by mnurulhuda** [comment]
I've tried to combine react-native-firebase with [geofirestore-js](https://github.com/MichaelSolati/geofirestore-js) but when searching for nearby users returns empty value.

```
import firestore from '@react-native-firebase/firestore';
import * as geofirestore from 'geofirestore';

const ref = firestore();
const GeoFirestore = geofirestore.initializeApp(ref);

const geocollection = GeoFirestore.collection("users");
const query = geocollection.near({ center: new firestore.GeoPoint(-6.719648, 111.346118), radius: 1000 });

query.get().then((value) => {
  console.log(value.size);
});
```

The code above returns the value of 0. While in Cloud Firestore the users data is less than 1 km.

**Reply by mikehardy** [reply]
I quote myself:

> It is possible to use both at the same time, so it is possible to for it to work, but you may have data coherency issues if both are attempting to cache things and have slightly different data. Or it might work great.

**Reply by mikehardy** [reply]
Please note we have nothing to do with geofirestore-js, so I will offer no support for that.
